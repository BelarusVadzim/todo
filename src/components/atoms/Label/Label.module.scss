@import 'src/variables.scss';

$regular-label-color: $secondary-font-color;
$lineThrough-label-color: $inactive-color;

.label {
  font-size: $font-size;
  color: $regular-label-color;
  border: none;
  font-family: $main-font;
  cursor: default;
  min-width: none;
}

.label_lineThrough {
  @extend .label;
  text-decoration: line-through;
  color: lighten($lineThrough-label-color, 10%);
}

