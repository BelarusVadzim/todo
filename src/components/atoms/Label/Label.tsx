import { PropsWithChildren } from "react";
import style from "./Label.module.scss";

type LabelProps = PropsWithChildren;

const Label: React.FC<LabelProps> = ({ children }) => {
  return <span className={style.label}>{children}</span>;
};

export default Label;
