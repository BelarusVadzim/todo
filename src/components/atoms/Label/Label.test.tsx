import render  from 'react-test-renderer';
import Label from './Label';

jest.mock('./Label.module.scss', () => ({ labelLineThrough: 'labelLineThroughstyle', label: 'labelStyle' }));

describe('<Label />', () => {
  it('should render properly', () => {
    const tree = render
      .create(<Label>LabelChild</Label>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when draggable lineThrough className specified should render properly', () => {
    const tree = render
      .create(<Label draggable lineThrough className='testClass'>LabelChild</Label>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
