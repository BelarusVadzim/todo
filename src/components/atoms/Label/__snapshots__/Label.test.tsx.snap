// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Label /> should render properly 1`] = `
<div
  className="labelStyle"
>
  LabelChild
</div>
`;

exports[`<Label /> when draggable lineThrough className specified should render properly 1`] = `
<div
  className="labelLineThroughstyle testClass"
  data-draggable="true"
>
  LabelChild
</div>
`;
